src/catkin_pkg/package_version.py

```diff
@@ -84,6 +84,28 @@ def _replace_version(package_str, new_version):
     return new_package_str
 
 
+def _replace_setup_py_version(setup_py_str, new_version):
+    """
+    Replace the version tag in contents if there is only one instance and it is using a literal as the version.
+
+    :param str package_str: contents of setup.py
+    :param str new_version: new version number
+    :returns: new setup.py string
+    :rtype: str
+    :raises RuntimeError:
+    """
+    # try to replace contens
+    new_setup_py_str, number_of_subs = re.subn(
+        r'version=([\'"])\d+\.\d+\.\d+([\'"]),',
+        r'version=\g<1>%s\g<2>,' % new_version,
+        setup_py_str)
+    if number_of_subs == 0:
+        raise RuntimeError("Failed to find a normal version statement, e.g.: version='1.2.3',")
+    if number_of_subs != 1:
+        raise RuntimeError('Illegal number of version statements: %s' % (number_of_subs))
```

Comment:
I appreciate that these are hard errors. I think it's the correct thing to do and since ament_package templates use literal version strings in setup.py hopefully there aren't too many packages out there that have gotten fancy.